### PR TITLE
Add Feature to check if any character has reached a specified rank

### PR DESCRIPTION
**Pull request by Mjb141**

### CODE REVIEW

**Comment by msolefonte** [comment]
Wait a sec on this one

**Comment by msolefonte** [comment]
I just rembered this exists (Ensure 2 wizards have reached level 10): 

```objective
{
	type ACHIEVE_CHARACTER_RANK;
	total 2;
	total2 10;
	agent wizard;
}
````

**Comment by msolefonte** [comment]
I do not think it works for lords tho.

**Comment by msolefonte** [comment]
Also, use `character:rank() >= level` and ensure you are only checking Skarband. For that check `character:unit_key()` equals `wh3_main_kho_skarbrand`

**Comment by Mjb141** [comment]
The event interface for `CharacterRankUp` is `CHARACTER_SCRIPT_INTERFACE` - does this mean `context` _is_ `character` and I only need to use `character:rank()` or should I still use `context:character():rank()`?

**Comment by msolefonte** [comment]
> The event interface for `CharacterRankUp` is `CHARACTER_SCRIPT_INTERFACE` - does this mean `context` _is_ `character` and I only need to use `character:rank()` or should I still use `context:character():rank()`?

Good question! `context` is a complex object. It is never a character or a faction or whatever, but a collection of items. Picture it as a dictionary where the getter methods are hardcoded to `character()`, `faction()` and such.

File: src/script/campaign/mod/vco-listeners.lua [hunk 1]
```diff
@@ -28,7 +28,7 @@ local function is_faction_vassal_or_destroyed(player_faction, target_faction_key
 end
 
 local function is_character_level_x(character, level)
```
Comment (msolefonte) [inline src/script/campaign/mod/vco-listeners.lua:30]: I am always up for descriptive functions, but maybe this one is too obvious and small? Up to you.

Comment (Mjb141) [reply src/script/campaign/mod/vco-listeners.lua:30]: Oh no, I very much like my functions to be appropriately named (i.e. immediately obvious) and short. This is fine in my book!

**Review by msolefonte — APPROVED** [review]

**Pull request merged**